pennylane/devices/tests/test_gates.py

```diff
@@ -358,15 +358,25 @@ def test_supported_gates_can_be_implemented(self, device_kwargs, operation):
         device_kwargs["wires"] = 4  # maximum size of current gates
         dev = qml.device(**device_kwargs)
 
-        assert hasattr(dev, "operations")
-        if operation in dev.operations:
+        if isinstance(dev, qml.Device):
+            assert hasattr(dev, "operations")
```

Comment:
I know you just kept what was there, but it's in the Device class definition, so this is kinda unnecessary
```suggestion
```